sdk/js/index.ts

```diff
@@ -167,32 +260,34 @@ export default class StreamManager {
       throw new Error("Error while processing trasactions");
 
     return new Stream(
-      this.address,
+      this,
       account,
       streamId,
+      token,
+      amountPerSecond,
       this.publicClient,
-      this.walletClient,
+      this.walletClient
     );
   }
 
   onStreamCreated(
     handleStream: (stream: Stream) => null,
-    creator?: Address,
+    creator?: Address
   ): void {
     const onLogs = (logs: Log[]) => {
       logs
         .map(
           // Log is StreamCreated
           (log: Log) =>
             new Stream(
```

Comment:
this should probably use `Stream.fromEventLog` instead

maybe as like `logs.map((log) => Stream.fromEventLog(...).then(handleStream).catch(console.error))`

Reply:
we are moving this to another PR